### PR TITLE
add sample

### PR DESCRIPTION
**Pull request by dan4ik605743**
I wrote a bot that can send weather and exchange rates based on your library for working with api telegram, don't you want it to be in the example? Perhaps people seeing it would be easier to work with this library.

### CODE REVIEW

**Review by reo7sp — CHANGES_REQUESTED** [review]
Commit this example in samples directory directly, not via submodule

**Comment by dan4ik605743** [comment]
> Commit this example in samples directory directly, not via submodule

So the code will change, for example, right now I have several commits that are not ready that add inline-keyboard through a separate compilation flag, maybe it's better to let it be a submodule, how to fix it.

**Comment by reo7sp** [comment]
No, sample must be complete and committed

**Comment by dan4ik605743** [comment]
> 



> No, sample must be complete and committed

Well, this is considered as a project rather than an example, but I would like to place it somewhere for you, as an example. People who would like to use this library and do not know where to start, it would certainly help and, in general, how their ideas can be implemented. Therefore, I don’t know, you can place it not in samples, in README.md give a link to the repository there is an exemplary project, what do you say? Because I'm improving the code / adding something new. So I can't copy all the files to samples/.

**Comment by dan4ik605743** [comment]
what do you say?

**Comment by dan4ik605743** [comment]
I want to help people who decide to use this library, to make some kind of contribution.

**Comment by reo7sp** [comment]
Extract from your project some isolated and simple code and commit this as an sample.

Sample as a huge project would be hard to understand, that’s why it won’t be useful for people. Sample should be clear, small and easy to compile

**Comment by dan4ik605743** [comment]
something simple cannot be singled out from the whole project, everything is connected with each other. I can make one example with which it would be easier, now I will write.

**Comment by dan4ik605743** [comment]
I would like to add this example. what do you say?

File: samples/received-text-processing/src/main.cpp [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        }
+
+        for (const auto& command : bot_commands) {
+            if ("/" + command == message->text) {
+                return;
+            }
+        }
+
+        bot.getApi().sendMessage(message->chat->id, "unknown command");
+    });
```
Comment (reo7sp) [inline samples/received-text-processing/src/main.cpp:50]: Let long_poll.start() be invoked only in the main loop.

```suggestion
    bool test_text_state = false;

    Bot bot(token);
    TgLongPoll long_poll(bot);

    bot.getEvents().onCommand("start", [&bot](Message::Ptr message) {
        bot.getApi().sendMessage(message->chat->id, "Hi!");
    });

    bot.getEvents().onCommand("test", [&](Message::Ptr message) {
        bot.getApi().sendMessage(message->chat->id, "Enter text");
        test_text_state = true;
    });

    bot.getEvents().onAnyMessage([&](TgBot::Message::Ptr message) {
        if (test_text_state) {
            bot.getApi().sendMessage(message->chat->id, message->text);
            test_text_state = false;
            return;
        }

        for (const auto& command : bot_commands) {
            if ("/" + command == message->text) {
                return;
            }
        }

        bot.getApi().sendMessage(message->chat->id, "unknown command");
    });
```

**Review by reo7sp — CHANGES_REQUESTED** [review]

File: samples/received-text-processing/src/main.cpp (hunk 1 above)
Comment (dan4ik605743) [reply samples/received-text-processing/src/main.cpp:50]: ? you have now changed the code it will not work as it should.

Comment (dan4ik605743) [reply samples/received-text-processing/src/main.cpp:50]: the point of all this is the second call to long_poll to get the text from the X input.

**Comment by dan4ik605743** [comment]
@reo7sp done

**Comment by dan4ik605743** [comment]
i rename branch

**Pull request closed without merging**